kuka_external_control_sdk/kss/include/kuka/external-control-sdk/kss/message_builder.h

```diff
@@ -23,43 +23,54 @@
 #include <string>
 
 #include "kuka/external-control-sdk/common/message_builder.h"
+#include "kuka/external-control-sdk/kss/gpio_value.h"
 
 namespace kuka::external::control::kss {
 
 class MotionState : public BaseMotionState {
- public:
-  MotionState(std::size_t dof) : BaseMotionState(dof) {
+public:
+  MotionState(std::size_t dof, std::vector<GPIOConfiguration> gpio_config_list)
+      : BaseMotionState(dof) {
     measured_positions_.resize(dof, std::numeric_limits<double>::quiet_NaN());
     measured_torques_.resize(dof, std::numeric_limits<double>::quiet_NaN());
     measured_velocities_.resize(dof, std::numeric_limits<double>::quiet_NaN());
-    measured_cartesian_positions_.resize(6, std::numeric_limits<double>::quiet_NaN());
+    measured_cartesian_positions_.resize(
+        6, std::numeric_limits<double>::quiet_NaN());
 
     first_cartesian_position_index_ += kMessagePrefix.length();
     first_cartesian_position_index_ += kCartesianPositionsPrefix.length() - 1;
-  }
 
-  MotionState(const MotionState& other) = default;
-  MotionState& operator=(const MotionState& other);
+    for (size_t i = 0; i < gpio_config_list.size(); i++) {

```

Comment:
does it make sense to move the shared ptr? the instance count will be decreased immediately after it goes out of scope